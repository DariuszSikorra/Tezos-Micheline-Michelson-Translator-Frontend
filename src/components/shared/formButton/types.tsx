export type ButtonProps = {
  type: 'submit' | 'reset' | 'button';
  label: string;
  className?: string;
  onClick?: any;
  disabled?: boolean;
  stylingType?: string;
};
