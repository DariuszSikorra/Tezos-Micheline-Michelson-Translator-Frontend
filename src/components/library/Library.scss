.Library {
  &__header {
    color: #ffffff;
    background: #787878;
    border-radius: 3px 3px 0 0;
    padding: 10px;
    height: 30px;
    font-family: Roboto;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    text-align: center;
    justify-content: center;
    letter-spacing: 0.15px;
    text-transform: uppercase;

    button {
      background: transparent;
      border: 0;
      border-bottom: 2px solid #787878;
      padding: 11px;
      font-family: Roboto;
      font-style: normal;
      font-weight: 500;
      font-size: 12px;
      line-height: 14px;
      display: flex;
      align-items: center;
      text-align: center;
      letter-spacing: 0.15px;
      text-transform: uppercase;
      color: #FFFFFF;
      outline: 0;

      &:hover {
        cursor: pointer;
        border-bottom: 2px solid #ffffff;
      }
    }

    img {
      display: flex;
    }

    &-selected {
        border-bottom: 2px solid #ff0000 !important;
    }

    &__translate-button {
      margin-left: 40px;
      margin-right: 40px;
      cursor: pointer;
    }

    @media screen and (max-width: 500px) {
      height: 150px;

      button, img {
        flex: 0 0 100%;
        flex-wrap: wrap;
        display: block;
      }
    }
  }
}
