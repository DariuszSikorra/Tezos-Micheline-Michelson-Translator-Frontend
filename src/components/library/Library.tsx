import React from "react";
import { useDispatch, useMappedState } from "redux-react-hook";
import "./Library.scss";
import LibraryItem from "./LibraryItem";
import { translate } from "../translator/Translator";
import * as TranslatorTypes from "../../store/translator/types";
import { IState } from "../../store/global/types";

const mapState = (state: IState) => ({
  library: state.library,
});

const Library = () => {
  const dispatch = useDispatch();
  const { library } = useMappedState(mapState);

  const selectMicheline = (value: string) => {
    dispatch({
      type: TranslatorTypes.TRANSLATOR_SET_MODE,
      mode: TranslatorTypes.Modes.MICHELINEMICHELSON,
    });

    dispatch({
      type: TranslatorTypes.TRANSLATOR_SET_MICHELINE,
      translation: value,
    });

    translate(TranslatorTypes.Modes.MICHELINEMICHELSON, value, dispatch);
  };

  const selectMichelson = (value: string) => {
    dispatch({
      type: TranslatorTypes.TRANSLATOR_SET_MODE,
      mode: TranslatorTypes.Modes.MICHELSONMICHELINE,
    });

    dispatch({
      type: TranslatorTypes.TRANSLATOR_SET_MICHELSON,
      translation: value,
    });

    translate(TranslatorTypes.Modes.MICHELSONMICHELINE, value, dispatch);
  };

  return (
    <div className="Library">
      {library.map((item, key) => (
        <LibraryItem
          selectMicheline={selectMicheline}
          selectMichelson={selectMichelson}
          data={item}
          key={`${item.source}${key.toString()}`}
        />
      ))}
    </div>
  );
};

export default Library;
