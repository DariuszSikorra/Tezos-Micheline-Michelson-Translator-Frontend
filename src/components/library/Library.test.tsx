import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { StoreContext } from "redux-react-hook";
import configureStore from "redux-mock-store";
import Library from "./Library";

import LibraryItem from "./LibraryItem";

import * as TranslatorActions from "../../store/translator/actions";
import * as TranslatorTypes from "../../store/translator/types";

const library = [
  {
    source:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
    translation:
      '[ { "prim": "parameter", "args": [ { "prim": "int" } ] }, { "prim": "storage", "args": [ { "prim": "int" } ] }, { "prim": "code", "args": [ [ { "prim": "CAR" },{ "prim": "PUSH", "args": [ { "prim": "int" }, { "int": "1" } ] },{ "prim": "ADD" },{ "prim": "NIL", "args": [ { "prim": "operation" } ] },{ "prim": "PAIR" } ] ] } ]\n',
  },
  {
    source:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
    translation:
      '[ { "prim": "parameter", "args": [ { "prim": "int" } ] }, { "prim": "storage", "args": [ { "prim": "int" } ] }, { "prim": "code", "args": [ [ { "prim": "CAR" },{ "prim": "PUSH", "args": [ { "prim": "int" }, { "int": "1" } ] },{ "prim": "ADD" },{ "prim": "NIL", "args": [ { "prim": "operation" } ] },{ "prim": "PAIR" } ] ] } ]\n',
  },
  {
    source:
      '[\n    {\n      "prim": "parameter",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "storage",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "code",\n      "args": [\n        [\n          {\n            "prim": "CAR"\n          },\n          {\n            "prim": "PUSH",\n            "args": [\n              {\n                "prim": "int"\n              },\n              {\n                "int": "1"\n              }\n            ]\n          },\n          {\n            "prim": "ADD"\n          },\n          {\n            "prim": "NIL",\n            "args": [\n              {\n                "prim": "operation"\n              }\n            ]\n          },\n          {\n            "prim": "PAIR"\n          }\n        ]\n      ]\n    }\n]',
    translation:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
  },
  { source: "EQ", translation: '{ "prim": "EQ" }\n' },
  {
    source:
      '[\n    {\n      "prim": "parameter",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "storage",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "code",\n      "args": [\n        [\n          {\n            "prim": "CAR"\n          },\n          {\n            "prim": "PUSH",\n            "args": [\n              {\n                "prim": "int"\n              },\n              {\n                "int": "1"\n              }\n            ]\n          },\n          {\n            "prim": "ADD"\n          },\n          {\n            "prim": "NIL",\n            "args": [\n              {\n                "prim": "operation"\n              }\n            ]\n          },\n          {\n            "prim": "PAIR"\n          }\n        ]\n      ]\n    }\n]',
    translation:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
  },
  {
    source:
      '[\n    {\n      "prim": "parameter",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "storage",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "code",\n      "args": [\n        [\n          {\n            "prim": "CAR"\n          },\n          {\n            "prim": "PUSH",\n            "args": [\n              {\n                "prim": "int"\n              },\n              {\n                "int": "1"\n              }\n            ]\n          },\n          {\n            "prim": "ADD"\n          },\n          {\n            "prim": "NIL",\n            "args": [\n              {\n                "prim": "operation"\n              }\n            ]\n          },\n          {\n            "prim": "PAIR"\n          }\n        ]\n      ]\n    }\n]',
    translation:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
  },
  {
    source:
      '[\n  {\n    "prim": "parameter",\n    "args": [\n      {\n        "prim": "int"\n      }\n    ]\n  },\n  {\n    "prim": "storage",\n    "args": [\n      {\n        "prim": "int"\n      }\n    ]\n  },\n  {\n    "prim": "code",\n    "args": [\n      [\n        {\n          "prim": "CAR"\n        },\n        {\n          "prim": "PUSH",\n          "args": [\n            {\n              "prim": "int"\n            },\n            {\n              "int": "1"\n            }\n          ]\n        },\n        {\n          "prim": "ADD"\n        },\n        {\n          "prim": "NIL",\n          "args": [\n            {\n              "prim": "operation"\n            }\n          ]\n        },\n        {\n          "prim": "PAIR"\n        }\n      ]\n    ]\n  }\n]',
    translation:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
  },
];

const mockStore = configureStore();

describe("Library", () => {
  let wrapper: ReactWrapper;
  let store: any;

  beforeEach(() => {
    store = mockStore({ library });

    store.dispatch = jest.fn();

    wrapper = mount(
      <StoreContext.Provider value={store}>
        <Library />
      </StoreContext.Provider>,
    );
  });

  it("renders without crashing", () => {
    const item = wrapper.find(LibraryItem);

    expect(
      item
        .at(0)
        .find(".LibraryItem__detail-wrapper--label")
        .at(0)
        .text(),
    ).toEqual("Name");

    expect(
      item
        .at(0)
        .find(".LibraryItem__detail-wrapper--content")
        .at(0)
        .text(),
    ).toEqual("some name");

    expect(item.at(0).props().data).toEqual(library[1]);
    expect(item).toHaveLength(7);
  });

  it("selecting Micheline source is working", () => {
    const item = wrapper.find(LibraryItem);

    const itemMicheline = item
      .at(0)
      .find(".LibraryItem__wrapper2")
      .at(0);

    itemMicheline.simulate("click");
    expect(store.dispatch).toHaveBeenCalledTimes(3);
    expect(store.dispatch).toHaveBeenCalledWith(
      TranslatorActions.TranslatorSetMode(TranslatorTypes.Modes.MICHELINEMICHELSON),
    );
    expect(store.dispatch).toHaveBeenCalledWith({
      type: TranslatorTypes.TRANSLATOR_SET_MICHELINE,
      translation: library[0].source,
    });
    expect(store.dispatch).toHaveBeenCalledWith(
      TranslatorActions.TranslatorFetchMichelineToMichelson(library[0].source),
    );
  });

  it("selecting Michelson source is working", () => {
    const item = wrapper.find(LibraryItem);

    const itemMicheline = item
      .at(0)
      .find(".LibraryItem__wrapper2")
      .at(1);

    itemMicheline.simulate("click");
    expect(store.dispatch).toHaveBeenCalledTimes(3);
    expect(store.dispatch).toHaveBeenCalledWith(
      TranslatorActions.TranslatorSetMode(TranslatorTypes.Modes.MICHELSONMICHELINE),
    );
    expect(store.dispatch).toHaveBeenCalledWith({
      type: TranslatorTypes.TRANSLATOR_SET_MICHELSON,
      translation: library[0].translation,
    });
    expect(store.dispatch).toHaveBeenCalledWith(
      TranslatorActions.TranslatorFetchMichelsonToMicheline(library[0].translation),
    );
  });

  it("matches snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
