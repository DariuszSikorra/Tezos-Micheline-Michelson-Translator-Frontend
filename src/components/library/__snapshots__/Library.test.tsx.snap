// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library matches snapshot 1`] = `
"<Library>
  <div className=\\"Library\\">
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Name
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              some name
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              some description ewr ewwer wer werwer wer wer werwer werwe rwe rewr wer werwe rwe rewrewr ewr ew ewrewrwer rew rwerwe rwerwerwerwe rwerwerwe rweewrwe wrwe we rwe werew rwe
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            [ { &quot;prim&quot;: &quot;parameter&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; } ] }, { &quot;prim&quot;: &quot;storage&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; } ] }, { &quot;prim&quot;: &quot;code&quot;, &quot;args&quot;: [ [ { &quot;prim&quot;: &quot;CAR&quot; },{ &quot;prim&quot;: &quot;PUSH&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; }, { &quot;int&quot;: &quot;1&quot; } ] },{ &quot;prim&quot;: &quot;ADD&quot; },{ &quot;prim&quot;: &quot;NIL&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;operation&quot; } ] },{ &quot;prim&quot;: &quot;PAIR&quot; } ] ] } ]
            
          </div>
        </div>
      </div>
    </LibraryItem>
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Name
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              some name
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              some description ewr ewwer wer werwer wer wer werwer werwe rwe rewr wer werwe rwe rewrewr ewr ew ewrewrwer rew rwerwe rwerwerwerwe rwerwerwe rweewrwe wrwe we rwe werew rwe
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            [ { &quot;prim&quot;: &quot;parameter&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; } ] }, { &quot;prim&quot;: &quot;storage&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; } ] }, { &quot;prim&quot;: &quot;code&quot;, &quot;args&quot;: [ [ { &quot;prim&quot;: &quot;CAR&quot; },{ &quot;prim&quot;: &quot;PUSH&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; }, { &quot;int&quot;: &quot;1&quot; } ] },{ &quot;prim&quot;: &quot;ADD&quot; },{ &quot;prim&quot;: &quot;NIL&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;operation&quot; } ] },{ &quot;prim&quot;: &quot;PAIR&quot; } ] ] } ]
            
          </div>
        </div>
      </div>
    </LibraryItem>
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Name
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              some name
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              some description ewr ewwer wer werwer wer wer werwer werwe rwe rewr wer werwe rwe rewrewr ewr ew ewrewrwer rew rwerwe rwerwerwerwe rwerwerwe rweewrwe wrwe we rwe werew rwe
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            [
                {
                  &quot;prim&quot;: &quot;parameter&quot;,
                  &quot;args&quot;: [
                    {
                      &quot;prim&quot;: &quot;int&quot;
                    }
                  ]
                },
                {
                  &quot;prim&quot;: &quot;storage&quot;,
                  &quot;args&quot;: [
                    {
                      &quot;prim&quot;: &quot;int&quot;
                    }
                  ]
                },
                {
                  &quot;prim&quot;: &quot;code&quot;,
                  &quot;args&quot;: [
                    [
                      {
                        &quot;prim&quot;: &quot;CAR&quot;
                      },
                      {
                        &quot;prim&quot;: &quot;PUSH&quot;,
                        &quot;args&quot;: [
                          {
                            &quot;prim&quot;: &quot;int&quot;
                          },
                          {
                            &quot;int&quot;: &quot;1&quot;
                          }
                        ]
                      },
                      {
                        &quot;prim&quot;: &quot;ADD&quot;
                      },
                      {
                        &quot;prim&quot;: &quot;NIL&quot;,
                        &quot;args&quot;: [
                          {
                            &quot;prim&quot;: &quot;operation&quot;
                          }
                        ]
                      },
                      {
                        &quot;prim&quot;: &quot;PAIR&quot;
                      }
                    ]
                  ]
                }
            ]
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
      </div>
    </LibraryItem>
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Name
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              some name
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              some description ewr ewwer wer werwer wer wer werwer werwe rwe rewr wer werwe rwe rewrewr ewr ew ewrewrwer rew rwerwe rwerwerwerwe rwerwerwe rweewrwe wrwe we rwe werew rwe
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            EQ
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            { &quot;prim&quot;: &quot;EQ&quot; }
            
          </div>
        </div>
      </div>
    </LibraryItem>
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Name
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              some name
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              some description ewr ewwer wer werwer wer wer werwer werwe rwe rewr wer werwe rwe rewrewr ewr ew ewrewrwer rew rwerwe rwerwerwerwe rwerwerwe rweewrwe wrwe we rwe werew rwe
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            [
                {
                  &quot;prim&quot;: &quot;parameter&quot;,
                  &quot;args&quot;: [
                    {
                      &quot;prim&quot;: &quot;int&quot;
                    }
                  ]
                },
                {
                  &quot;prim&quot;: &quot;storage&quot;,
                  &quot;args&quot;: [
                    {
                      &quot;prim&quot;: &quot;int&quot;
                    }
                  ]
                },
                {
                  &quot;prim&quot;: &quot;code&quot;,
                  &quot;args&quot;: [
                    [
                      {
                        &quot;prim&quot;: &quot;CAR&quot;
                      },
                      {
                        &quot;prim&quot;: &quot;PUSH&quot;,
                        &quot;args&quot;: [
                          {
                            &quot;prim&quot;: &quot;int&quot;
                          },
                          {
                            &quot;int&quot;: &quot;1&quot;
                          }
                        ]
                      },
                      {
                        &quot;prim&quot;: &quot;ADD&quot;
                      },
                      {
                        &quot;prim&quot;: &quot;NIL&quot;,
                        &quot;args&quot;: [
                          {
                            &quot;prim&quot;: &quot;operation&quot;
                          }
                        ]
                      },
                      {
                        &quot;prim&quot;: &quot;PAIR&quot;
                      }
                    ]
                  ]
                }
            ]
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
      </div>
    </LibraryItem>
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Name
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              some name
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              some description ewr ewwer wer werwer wer wer werwer werwe rwe rewr wer werwe rwe rewrewr ewr ew ewrewrwer rew rwerwe rwerwerwerwe rwerwerwe rweewrwe wrwe we rwe werew rwe
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            [
                {
                  &quot;prim&quot;: &quot;parameter&quot;,
                  &quot;args&quot;: [
                    {
                      &quot;prim&quot;: &quot;int&quot;
                    }
                  ]
                },
                {
                  &quot;prim&quot;: &quot;storage&quot;,
                  &quot;args&quot;: [
                    {
                      &quot;prim&quot;: &quot;int&quot;
                    }
                  ]
                },
                {
                  &quot;prim&quot;: &quot;code&quot;,
                  &quot;args&quot;: [
                    [
                      {
                        &quot;prim&quot;: &quot;CAR&quot;
                      },
                      {
                        &quot;prim&quot;: &quot;PUSH&quot;,
                        &quot;args&quot;: [
                          {
                            &quot;prim&quot;: &quot;int&quot;
                          },
                          {
                            &quot;int&quot;: &quot;1&quot;
                          }
                        ]
                      },
                      {
                        &quot;prim&quot;: &quot;ADD&quot;
                      },
                      {
                        &quot;prim&quot;: &quot;NIL&quot;,
                        &quot;args&quot;: [
                          {
                            &quot;prim&quot;: &quot;operation&quot;
                          }
                        ]
                      },
                      {
                        &quot;prim&quot;: &quot;PAIR&quot;
                      }
                    ]
                  ]
                }
            ]
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
      </div>
    </LibraryItem>
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Name
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              some name
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Date created
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              2019-01-01
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              some description ewr ewwer wer werwer wer wer werwer werwe rwe rewr wer werwe rwe rewrewr ewr ew ewrewrwer rew rwerwe rwerwerwerwe rwerwerwe rweewrwe wrwe we rwe werew rwe
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            [
              {
                &quot;prim&quot;: &quot;parameter&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;storage&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;code&quot;,
                &quot;args&quot;: [
                  [
                    {
                      &quot;prim&quot;: &quot;CAR&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;PUSH&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;int&quot;
                        },
                        {
                          &quot;int&quot;: &quot;1&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;ADD&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;NIL&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;operation&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;PAIR&quot;
                    }
                  ]
                ]
              }
            ]
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
      </div>
    </LibraryItem>
  </div>
</Library>"
`;
